txnsync/interfaces.go

```diff
@@ -71,7 +71,12 @@ type NodeConnector interface {
 	Random(uint64) uint64
 	GetPeers() []PeerInfo
 	GetPeer(interface{}) PeerInfo // get a single peer given a network peer opaque interface
-	UpdatePeers([]*Peer, []interface{})
+	// UpdatePeers call is being made to inform the node that either a link need to be established
+	// between the set of the txsyncPeers peers and the set of netPeers, or that the peersAverageDataExchangeRate
+	// was recalculated and could potentially be updated.
+	// The peersAverageDataExchangeRate passed in here is the average communication rate ( measured in bytes per second )
+	// across all the connected peers.
+	UpdatePeers(txsyncPeers []*Peer, netPeers []interface{}, peersAverageDataExchangeRate uint64)
```

Comment:
Sorry but this still isn't clear to me, unless I'm mistaking something.

I see `averageDataExchangeRate /= uint64(len(peersInfo))`, so it looks like this isn't the rate across all peers but the rate per peer.

Reply:
it's the average..